src/datarobotx/idp/guard_configurations.py

```diff
@@ -0,0 +1,324 @@
+        for config in guard_config:
+            if guard_token in config.get("description", ""):
+                return str(latest_version_id)
+
+    cleaned_guard_config = _clean_guard_configurations(guard_config)
+
+    if replace:
+        # delete the existing guard configuration if it exists
+        cleaned_guard_config = [
+            config for config in cleaned_guard_config if guard_name != config["name"]
+        ]
+
+    # get the guard templates
+    guard_templates = client.get("guardTemplates/").json()["data"]
+    guard_templates = [
+        from_api(gt) for gt in guard_templates if gt["name"] == guard_config_template_name
+    ]
+
+    if not guard_templates:
+        raise ValueError(f"Guard template {guard_config_template_name} not found")
+
+    selected_template = guard_templates[0]
+    if not all([stage in selected_template["allowed_stages"] for stage in stages]):
+        raise ValueError(
+            f"Guard template {guard_config_template_name} does not support stages {stages}"
+        )
+
+    # Assemble the guard configuration with all expected fields
+    cleaned_guard_template = _clean_guard_configurations([selected_template])[0]
+
+    cleaned_guard_template[
+        "description"
+    ] = f"{description or cleaned_guard_template['description']} [{guard_token}]"
+    cleaned_guard_template["stages"] = [stage for stage in stages]
+    cleaned_guard_template["intervention"] = intervention
+    cleaned_guard_template.update(guard_config_template_settings)
+    cleaned_guard_template["name"] = guard_name
+
+    # append previous guard configurations
+    cleaned_guard_config.append(cleaned_guard_template)
+    try:
+        res = client.post(
+            "guardConfigurations/toNewCustomModelVersion/",
+            json={
+                "data": [to_api(config) for config in cleaned_guard_config],
+                "customModelId": custom_model_id,
+            },
+        )
+    except ClientError:
+        _ = get_or_create_custom_model_version_from_previous(
+            endpoint=endpoint,
+            token=token,
+            custom_model_id=custom_model_id,
+            base_environment_id=base_environment_id,
+        )
+        res = client.post(
+            "guardConfigurations/toNewCustomModelVersion/",
+            json={
+                "data": [to_api(config) for config in cleaned_guard_config],
+                "customModelId": custom_model_id,
+            },
+        )
+    return str(res.json()["customModelVersionId"])
+
+
+def get_or_create_guard_config_to_custom_model_version(  # noqa: PLR0913
+    endpoint: str,
+    token: str,
+    custom_model_id: str,
+    guard_config_template_name: str,
+    guard_config_template_settings: Dict[str, Any],
+    stages: List[Union[Literal["prompt"], Literal["response"]]],
+    intervention: Intervention,
+    name: Optional[str] = None,
+    description: Optional[str] = None,
+) -> str:
+    """Add a guard configuration to a custom model version.
```

Comment:
Was assuming this would be something like "Get or create a custom model version with the requested guard configuration."